app/components/Form/ToggleSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { Flex } from '@webkom/lego-bricks';
+import cx from 'classnames';
+import { Keyboard } from 'app/utils/constants';
+import { createField } from './Field';
+import styles from './ToggleSwitch.module.css';
+import type {
+  ComponentProps,
+  InputHTMLAttributes,
+  KeyboardEvent,
+  MouseEvent,
+} from 'react';
+import type { Overwrite } from 'utility-types';
+
+type Props = {
+  className?: string;
+} & Overwrite<
+  InputHTMLAttributes<HTMLInputElement>,
+  { onChange?: (checked: boolean) => void }
+>;
+
+const ToggleSwitch = ({
+  id,
+  checked,
+  value,
+  className,
+  disabled,
+  ...props
+}: Props) => {
```

Comment:
It may be worth looking at the `ToggleButton` from `react-aria-components`. It may simplify this component a bit:)
https://react-spectrum.adobe.com/react-aria/ToggleButton.html

Reply:
Clever! I refactored it now - ended up much cleaner and "safer"